bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -1008,6 +1037,8 @@ public void run() {
                                 if (entry != null && (!syncData || entry.ackBeforeSync)) {
                                     toFlush.set(i, null);
                                     numEntriesToFlush--;
+                                    flushedNotSyncedEntries.add(
+                                            new AbstractMap.SimpleImmutableEntry<>(entry.ledgerId, entry.entryId));
```

Comment:
AbstractMap.SimpleImmutableEntry is to be replaced with a more specific class without boxing. Maybe it is better to use a Reclyclable object